pkg/reconciler/switchexternalcluster/reconciler.go

```diff
@@ -0,0 +1,94 @@
+package switchexternalcluster
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/log"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/postgres"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/reconciler/switchexternalcluster/common"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
+)
+
+// Reconcile reconciles the cluster hibernation status.
+func Reconcile(
```

Comment:
Making the operator lead the transition makes the implementation more complex but improves the stability